src/Raven.Server/Rachis/Follower.cs

```diff
@@ -672,36 +672,76 @@ private unsafe bool ReadSnapshot(SnapshotReader reader, ClusterOperationContext
                         return true;
                     case RootObjectType.VariableSizeTree:
                         size = reader.ReadInt32();
-                        reader.ReadExactly(size);
+                        reader.ReadExactly(size); 
 
                         Tree tree = null;
+                        Slice.From(context.Allocator, reader.Buffer, 0, size, ByteStringType.Immutable, out Slice treeName); // The Slice will be freed on context close
+
+                        entries = reader.ReadInt64();
+                        var flags = TreeFlags.FixedSizeTrees;
+
                         if (dryRun == false)
                         {
-                            Slice.From(context.Allocator, reader.Buffer, 0, size, ByteStringType.Immutable, out Slice treeName); // The Slice will be freed on context close
                             txw.DeleteTree(treeName);
                             tree = txw.CreateTree(treeName);
                         }
 
-                        entries = reader.ReadInt64();
+                        if (_connection.Features.MultiTree)
+                            flags = (TreeFlags)reader.ReadInt32();
+
                         for (long i = 0; i < entries; i++)
                         {
                             token.ThrowIfCancellationRequested();
+                            // read key
                             size = reader.ReadInt32();
                             reader.ReadExactly(size);
+
                             using (Slice.From(context.Allocator, reader.Buffer, 0, size, ByteStringType.Immutable, out Slice valKey))
                             {
-                                size = reader.ReadInt32();
-                                reader.ReadExactly(size);
-
-                                if (dryRun == false)
+                                switch (flags)
                                 {
-                                    using (tree.DirectAdd(valKey, size, out byte* ptr))
-                                    {
-                                        fixed (byte* pBuffer = reader.Buffer)
+                                    case TreeFlags.None:
+                                    case TreeFlags.FixedSizeTrees:
+                                    case TreeFlags.LeafsCompressed:
```

Comment:
Few things to note here and make sure I got that right..

1. Although those are marked as flags as far as I can see we don't treat it as a bit flag, expect for `TreeFlags.Streams` which we don't use in raft, once we will it will throw here.
2. The `LeafsCompressed` is only used when the env is compacted. Don't know if we can compact the system storage, so not sure what will happened here. But this is the exact behavior for this flag as previous impl.
3. The `MultiValue` can't be on its own and is a secondary tree to the `MultiValueTrees` so we should hit it here.
4. `None` is a _valid_ flag

Reply:
1. We have two usages as the flag 
 - `TreeFlags.Streams` here:
https://github.com/ravendb/ravendb/blob/ef53f2f4754c4458021bef21764bc0479ea725ea/src/Voron/Data/BTrees/Tree.Stream.cs#L231

- `TreeFlags.MultiValueTrees` here:
https://github.com/ravendb/ravendb/blob/ef53f2f4754c4458021bef21764bc0479ea725ea/src/Voron/Data/BTrees/Tree.MultiTree.cs#L49

But in case of `TreeFlags.MultiValueTrees` I don't see that we mix it with other flags.

2. `TreeFlags.LeafsCompressed` is _only_ used in map-reduce indexes. More precisely in trees where we store the map results (only on 64bits):

https://github.com/ravendb/ravendb/blob/ef53f2f4754c4458021bef21764bc0479ea725ea/src/Raven.Server/Documents/Indexes/MapReduce/MapReduceResultsStore.cs#L62

I think we should throw since we don't expect to get such trees here.

3. Yes, `TreeFlags.MultiValue` means that it's a secondary tree.
4. `TreeFlags.None` is just regular b+tree without any extra special behavior

Reply:
`LeafsCompressed` is relevant only for map/reduce, shouldn't be applied here.

Reply:
about the `LeafCompressed` we have this:
https://github.com/ravendb/ravendb/blob/d144ccb6871f805efaa9e4d5727a0891cae4e37d/src/Voron/Impl/Compaction/StorageCompaction.cs#L285

Reply:
but yes it seems to be relevant only for map-reduce